client/src/components/MyProfile/SpeakerPage.js

```diff
@@ -328,25 +330,84 @@ const SessionHeading = styled(Heading4)`
 
 const StatusAndMoreContainer = styled.div`
   display: flex;
-  flex-direction: column;
-  justify-content: space-between;
-
+  flex-wrap: wrap;
+  justify-content: flex-end;
+  align-content: center;
+  ${mediaQueryMin.m`
+    justify-content: space-between;
+  `}
+  ${mediaQueryMin.l`
+    justify-content: flex-end;
+  `}
   ${mediaQueryMin.xl`
-    flex-direction: row;
+    justify-content: space-between;
   `}
 `;
 
-const SessionStatus = styled(SessionHeading)`
-  display: block;
+const statusColors = status => {
+  const {
+    color: {
+      session_status_proposed,
+      session_status_accepted,
+      session_status_decline,
+      session_status_not_found,
+    }
+  } = theme;
+
+  const colorsDict = {
+    'proposed': session_status_proposed,
+    'accepted': session_status_accepted,
+    'withdrawn': session_status_decline,
+    'rejected': session_status_decline,
+  };
+  return colorsDict[status] || session_status_not_found;
+};
+
+const SessionStatusBadge = styled.p`
+  ${ ({ theme: { space, color, font }, status }) => `
+    max-height: 55px;
+    max-width: fit-content;
+
+    padding: ${space.m};
+    margin-bottom: ${space.m};
+
+    color: ${color.text_1};
+    background: ${statusColors(status) || color.session_status_not_found};
+
+    font-weight: ${font.weight_bold};
+    font-size: ${font.size_md};
+    border: double 5px ${color.border_2};
+
+    text-align: center;
+  `}
 `;
 
 const ToSessionLink = styled(InvertedButtonStyledLink)`
-  min-width: max-content;
-  align-self: flex-end;
+  ${({theme: { space } }) => `
+    height: 50px;
+    margin-right: ${space.s};
+    padding-top: ${space.l};
+    min-width: max-content;
+    align-self: flex-end;
+  `}
 `;
 
 // React components
 
+const SessionStatus = status => {
+  const messagesDict = {
+    'proposed': 'Waiting for review',
+    'accepted': 'Accepted',
+    'withdrawn': 'Session withdrawn',
+    'rejected': 'Sadly not this time',
+  };
```

Comment:
OK, 
1. Small thing: naming should be messagesDictionary 
2. Big thing: since we have two dictionaries (line 353), lets move them both to the top of the file right bellow imports and uppercase them, `STATUS_TO_COLOR_DICTIONARY` and `STATUS_TO_MESSAGE_DICTIONARY`